src/main/kotlin/org/jitsi/rtp/rtcp/rtcpfb/transport_layer_fb/tcc/RtcpFbTccPacket.kt

```diff
@@ -17,20 +17,204 @@
+import org.jitsi.rtp.rtcp.rtcpfb.transport_layer_fb.tcc.RtcpFbTccPacket.Companion.kMaxSizeBytes
+import org.jitsi.rtp.rtcp.rtcpfb.transport_layer_fb.tcc.RtcpFbTccPacket.Companion.kTimeWrapPeriodUs
+import org.jitsi.rtp.rtcp.rtcpfb.transport_layer_fb.tcc.RtcpFbTccPacket.Companion.kTransportFeedbackHeaderSizeBytes
+import org.jitsi.rtp.rtp.RtpSequenceNumber
+import org.jitsi.rtp.util.BufferPool
+import org.jitsi.rtp.util.RtpUtils
 import org.jitsi.rtp.util.get3BytesAsInt
 import org.jitsi.rtp.util.getByteAsInt
 import org.jitsi.rtp.util.getShortAsInt
-import java.util.TreeMap
+
+// Size in bytes of a delta time in rtcp packet.
+// Valid values are 0 (packet wasn't received), 1 or 2.
+typealias DeltaSize = Int
+
+class ReceivedPacket(val seqNum: Int, val deltaTicks: Short) {
+    operator fun component1(): Int = seqNum
+    operator fun component2(): Short = deltaTicks
+}
 
 /**
+ * This class is a port of TransportFeedback in
+ * transport_feedback.h/transport_feedback.cc in Chrome
+ * https://cs.chromium.org/chromium/src/third_party/webrtc/modules/rtp_rtcp/source/rtcp_packet/transport_feedback.h?l=95&rcl=20393ee9b7ba622f254908646a9c31bf87349fc7
+ *
+ * Because of this, it explicitly does NOT try to conform
+ * to Kotlin style or idioms, instead striving to match the
+ * Chrome code as closely as possible in an effort to make
+ * future updates easier.
+ */
+class RtcpFbTccPacketBuilder(
+    val rtcpHeader: RtcpHeaderBuilder = RtcpHeaderBuilder(),
+    var mediaSourceSsrc: Long = -1,
+    val feedbackPacketSeqNum: Int = -1
+) {
+    var base_seq_no_: RtpSequenceNumber = RtpSequenceNumber.INVALID
+        private set
+    // The reference time, in ticks.  Chrome passes this into BuildFeedbackPacket, but we don't
+    // hold the times in the same way, so we'll just assign it the first time we see
+    // a packet in AddReceivedPacket
+    private var base_time_ticks_: Long = -1
+    // The amount of packets_ whose status are represented
+    var num_seq_no_ = 0
+        private set
+    // The current chunk we're 'filling out' as packets
+    // are received
+    private var last_chunk_ = LastChunk()
+    // All but last encoded packet chunks.
+    private val encoded_chunks_ = mutableListOf<Chunk>()
+    // The size of the entire packet, in bytes
+    private var size_bytes_ = kTransportFeedbackHeaderSizeBytes
+    private var last_timestamp_us_: Long = 0
+    private val packets_ = mutableListOf<ReceivedPacket>()
+
+    fun SetBase(base_sequence: Int, ref_timestamp_us: Long) {
+        base_seq_no_ = RtpSequenceNumber(base_sequence)
+        base_time_ticks_ = (ref_timestamp_us % kTimeWrapPeriodUs) / kBaseScaleFactor
+        last_timestamp_us_ = GetBaseTimeUs()
+    }
+
+    fun AddReceivedPacket(seqNum: Int, timestamp_us: Long): Boolean {
+        val sequence_number = RtpSequenceNumber(seqNum)
+        var delta_full = (timestamp_us - last_timestamp_us_) % kTimeWrapPeriodUs
+        if (delta_full > kTimeWrapPeriodUs / 2) {
```

Comment:
I don't understand this. I thought this check is meant to convert from the `(-x , x)` space to `(-x/2, x/2)`, but it only checks one side. So `delta_full` could be anywhere in `(-kTimeWrapPeriodUs, kTimeWrapPeriodUs/2]`